mqtt/mqtt-bridge/src/persist/mod.rs

```diff
@@ -1,36 +1,28 @@
-use std::{error::Error, sync::Arc};
+use bincode::ErrorKind;
+use rocksdb::Error;
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
 
-use async_trait::async_trait;
-use futures_util::stream::Stream;
-use mqtt3::proto::Publication;
-use parking_lot::Mutex;
-
-mod memory;
-
-/// Persistence used in bridge.
-/// Elements are added, then can be removed once retrieved by the loader
-/// If one attempts to remove added elements before reading via the loader remove returns None
-#[async_trait]
-trait Persist<'a> {
-    type Loader: Stream;
-    type Error: Error;
-
-    async fn new(batch_size: usize) -> Self;
-
-    async fn push(&mut self, message: Publication) -> Result<Key, Self::Error>;
-
-    async fn remove(&mut self, key: Key) -> Option<Publication>;
-
-    async fn loader(&'a mut self) -> Arc<Mutex<Self::Loader>>;
-}
+pub mod loader;
+pub mod persistor;
+pub mod waking_state;
```

Comment:
These should all be pub right?

Reply:
For context, the way I structured the loader / persistor is that they take a StreamWakeableState implementation for their new() method.

Reply:
Anca and I discussed and we thought of a better approach. If we construct the waking_state implementation from inside the persistor, then we don't need to expose it (since it is just an implementation detail leaking out).

I will add a setting for use persistence. Based on this setting I will construct the appropriate waking_state. I am working on a separate PR for handling all persistence settings, so I am fine merging it as is.

Reply:
Actually, I disagree with myself. I will change it now.

Reply:
you can re-import as `pub` only those types as required, instead of making all modules public